src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File fromFile = createFile(fromFileName);
+        File toFile = createFile(toFileName);
+        String[] strings = getReadLines(fromFile);
+        int supplyAmount = accountAmount(strings, SUPPLY);
+        int buyAmount = accountAmount(strings, BUY);
+        int resultAmount = supplyAmount - buyAmount;
+        String report = createReport(supplyAmount, buyAmount, resultAmount);
+        writeToFile(toFile, report);
+    }
+
+    private File createFile(String fileName) {
+        return new File(fileName);
+    }
+
+    private int accountAmount(String[] strings, String typeOfAmount) {
+        int amount = 0;
+        for (String line : strings) {
+            String[] typeAndAmount = line.split(COMMA);
+            if (typeAndAmount.length == 2 && typeOfAmount.equals(SUPPLY)
+                    && typeAndAmount[0].equals("supply")) {
+                amount += Integer.parseInt(typeAndAmount[1]);
+            }
+            if (typeAndAmount.length == 2 && typeOfAmount.equals(BUY)
+                    && typeAndAmount[0].equals("buy")) {
+                amount += Integer.parseInt(typeAndAmount[1]);
+            }
+        }
+        return amount;
+    }
+
+    private String[] getReadLines(File fromFile) {
```

Comment:
weird method name - why not `readData`?

also try to place methods inside ot your classes in invocation order, i.e:

- if method A was invoked before method B, then method A should be above method B in the class

this way readers will read your code like a book :)